### DISCUSSION TITLE
Add keyboard shortcut to merge "Remote" into current branch

### DISCUSSION THREAD

**Discussion by MisterAI** [gitextensions/gitextensions#9857] [Ideas] [answered]
### Feature description

When merging branches in a repo with submodules I regularly need to merge "Remote" into "Local" in the submodules. In order to speed up this process and to avoid using the mouse for this I would like to have a keyboard shortcut for automatically merging in "Remote" into my current branch/commit.
![Screenshot 2022-02-02 080202](https://user-images.githubusercontent.com/15369936/152109235-d1bbcac8-549e-452b-b8dc-baa8fc37d393.png)

### Environment

- Git Extensions 3.5.4.12724
- Build 65f01f39982ccb121c4df7d4b00b506939553958
- Git 2.34.1.windows.1
- Microsoft Windows NT 10.0.18363.0
- .NET Framework 4.8.4420.0
- DPI 96dpi (no scaling)

**Comment by RussKie** [comment, marked as answer]
You can create a script to perform what you need, show it in the revision grid, and have a hotkey bound to it for quick invocation.
![image](https://user-images.githubusercontent.com/4403806/152628488-f3c92ef2-7637-471a-b488-401a006a9ffc.png)
![image](https://user-images.githubusercontent.com/4403806/152628492-8522301a-5b02-482c-b227-895194c71e8b.png)

**Comment by MisterAI** [comment]
@RussKie Thank you for your answer. This would be an option for me but I don't know how to reference "Remote" in the script. Is there a tag for that?

**Reply by mstv** [reply]
![image](https://user-images.githubusercontent.com/36601201/152702966-2446815f-e060-43bb-b976-4dfbf9c6319f.png)

**Reply by MisterAI** [reply]
Thanks for this hint. I tried that one out but for me e. g. `git merge {sRemote}` resolves to `git merge origin` (which in turn resolves to `git merge origin/HEAD`). This is however not what I intend to do. The "Remote" commit that I want to merge into the current branch is the commit that is pointed to by the branch in the superproject which I'm currently merging into my current branch in the superproject (as marked by the orange "Remote" tag in my first screenshot). Is there a way to reference this one as well?

**Reply by mstv** [reply]
You could do it manually and check the resulting command in GE's Git Command Log (hotkey `F12`).
Then try the "...Branch..." tags perhaps.
If the existing tags do not suffice, please create public example repos and elaborate what exactly shall be merged where.
(... and please use more detailed names. "remote" usually means the / a remote *repo*.)
